specs/_features/eip7732/beacon-chain.md

```diff
@@ -543,24 +858,52 @@ def process_execution_payload_header(state: BeaconState, block: BeaconBlock) ->
     signed_header = block.body.signed_execution_payload_header
     assert verify_execution_payload_header_signature(state, signed_header)
 
-    # Check that the builder is active non-slashed has funds to cover the bid
     header = signed_header.message
     builder_index = header.builder_index
     builder = state.validators[builder_index]
     assert is_active_validator(builder, get_current_epoch(state))
     assert not builder.slashed
     amount = header.value
-    assert state.balances[builder_index] >= amount
+    # Check that the builder is registered as a builder unless self-building with zero value
+    if not is_builder(builder):
+        assert builder_index == block.proposer_index
+        assert amount == 0
+
+    # Check that the builder is active non-slashed has funds to cover the bid
```

Comment:
```suggestion
    # Check that the builder is active, non-slashed, and has funds to cover the bid
```

But also, the comment is a little misleading since we check the first two before the comment.